config/slist.c

```diff
@@ -218,6 +218,89 @@ static intptr_t slist_native_get(const struct ConfigSet *cs, void *var,
   return (intptr_t) list;
 }
 
+/**
+ * number_string_plus_equals - Add to a Slist by string - Implements ConfigSetType::string_plus_equals()
+ */
+static int slist_string_plus_equals(const struct ConfigSet *cs, void *var,
+                                     const struct ConfigDef *cdef,
+                                     const char *value, struct Buffer *err)
+{
+  if (!cs || !cdef)
+    return CSR_ERR_CODE; /* LCOV_EXCL_LINE */
+
+  int rc = CSR_SUCCESS;
+
+  /* Store empty strings as NULL */
+  if (value && (value[0] == '\0'))
+    return rc |= CSR_SUC_NO_CHANGE; // return no change
+
+  struct Slist *list = *(struct Slist **) var;
+
+  if (slist_is_member(list, value))
+    return rc |= CSR_SUC_NO_CHANGE; // return no change
+  else
+    if (!list)
+      list = slist_parse(value, cdef->type);
+    else
+      slist_add_string(list, value); // add value to list
+
+  if (cdef->validator)
+  {
+    rc = cdef->validator(cs, cdef, (intptr_t) list, err);
+
+    if (CSR_RESULT(rc) != CSR_SUCCESS)
+    {
+      slist_free(&list);
```

Comment:
When validation fails, if the list has:
- One item, we should free it
- Two or more, we only want to remove the tail item

Reply:
slist_dup() at the beginning of the function. Returning modified list if it passes through validator, or the original if it doesn't.

Reply:
yes